[BugFix] Reset index information when creating tablet in rollup job

srlch opened the pull request:
## Why I'm doing:
When creating tablet in rollup job, the tablet has index information which is the same as the base table even if the rollup schema does not contain the index column. This will cause link inverted index files failed because no ivt files will be generated.

## What I'm doing:
Reset index information based on the rollup schema when creating tablet in rollup job.

Fixes #issue
https://github.com/StarRocks/StarRocksTest/issues/7905

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.4
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0

xiangguangyxg commented on fe/fe-core/src/main/java/com/starrocks/alter/RollupJobV2.java:
```diff
@@ -737,6 +738,20 @@ protected void runRunningJob() throws AlterCancelException {
         this.span.end();
     }
 
+    public static List<Index> getRollupIndexes(List<Index> indexes, List<Column> rollupSchema) {
+        List<Index> rollupIndexes = Lists.newArrayList();
+        Set<ColumnId> columnIdsSetForRollupSchema =
+                        rollupSchema.stream().map(col -> col.getColumnId()).collect(Collectors.toSet());
+
+        for (Index index : indexes) {
+            Set<ColumnId> columnIdsSetForIndex = index.getColumns().stream().collect(Collectors.toSet());
+            if (columnIdsSetForRollupSchema.containsAll​(columnIdsSetForIndex)) {
+                rollupIndexes.add(index);
+            }
+        }
+        return rollupIndexes;
+    }
+
```
better to add a member function of olap table ？

srlch replied:
Updated, add static function instead of member function in olap table, because the context does not set in olap table when set the index information in create tablet request

srlch committed:
[BugFix] Reset index information when creating tablet for rollup job tablet

## Why I'm doing:
When creating tablet in rollup job, the tablet has index information which is the same as the base table
even if the rollup schema does not contain the index column. This will cause link inverted index files
failed because no ivt files will be generated.

## What I'm doing:
Reset index information based on the rollup schema when creating tablet in rollup job.

Signed-off-by: srlch <linzichao@starrocks.com>

srlch committed:
fix

Signed-off-by: srlch <linzichao@starrocks.com>
